framework/web/HeaderCollection.php

```diff
@@ -163,16 +170,25 @@ public function remove($name)
     public function removeAll()
     {
         $this->_headers = [];
+        $this->_originalHeaderNames = [];
     }
 
     /**
      * Returns the collection as a PHP array.
      * @return array the array representation of the collection.
      * The array keys are header names, and the array values are the corresponding header values.
+     * Since 2.0.45 you can pass true here to get the headers list of original header names (case-sensitive) instead of
+     * default normalized (case-insensitive) ones.
      */
-    public function toArray()
+    public function toArray($originalNames = false)
```

Comment:
This change breaks BC - if someone overwrites this method he will get fatal error after this change.

Reply:
@bizley would you please add it to UPGRADE?

Reply:
Why we're introducing such BC break in the first place? We could add another method.

Reply:
@samdark I agree with @rob006 - it's a method signature change for a public method, we should not do that.

Reply:
Yes, right, why haven't I think of that. I'll add a new one.